ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/api/useWatchLastSnapshotForReports.ts

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { fetchReportHistory } from 'services/ReportsService';
+import { ReportConfiguration, ReportSnapshot } from 'services/ReportsService.types';
+import useInterval from 'hooks/useInterval';
+import { getRequestQueryString } from './apiUtils';
+import { getErrorMessage } from '../errorUtils';
+
+type Result = {
+    reportSnapshots: Record<string, ReportSnapshot>;
+    isLoading: boolean;
+    error: string | null;
+};
+
+export type FetchLastSnapshotReturn = Result & {
+    fetchSnapshots: () => void;
+};
+
+async function fetchLastReportJobForConfiguration(
```

Comment:
Note: need to port updates from [here](https://github.com/stackrox/stackrox/pull/7449/files#diff-0f33326ac776627b721ece3f2186582246d1d8436f1c4235dfc88dd47e88fe15) once that PR is merged.